test/evaluate.js

```diff
@@ -55,42 +55,33 @@ describe('evaluate', async function () {
     }
     /** @returns {Promise<RoundDetails>} */
     const fetchRoundDetails = async () => ({ ...SPARK_ROUND_DETAILS, retrievalTasks: [VALID_TASK] })
-    const setScoresCalls = []
-    const ieContractWithSigner = {
-      async setScores (roundIndex, participantAddresses, scores) {
-        setScoresCalls.push({ roundIndex, participantAddresses, scores })
-        return { hash: '0x234' }
-      },
+    const submitScoresCalls = []
+    const submitScores = async (participantAddresses, scores) => {
+      submitScoresCalls.push({ participantAddresses, scores })
+    }
```

Comment:
Can you preserve the name `setScoresCall` to reduce the code churn?

I am surprised we don't need to include the `roundIndex` when submitting scores. Could the lack of `roundIndex` make debugging more difficult for us?

How do you feel about keeping the `setScores` signature in the interfaces, i.e. modify `submitScores` to require `(roundIndex, participantAddresses, scores)`, even if the actual implementation calling spark-rewards does not use the round index? 

Since it's something used by the previous version, I think we may need to have it again in the future. I know it's just a speculation violating YAGNI principle, so 🤷🏻  What I like most is that if we preserve the signature then there will be much fewer changes in the existing tests.

Reply:
> Can you preserve the name setScoresCall to reduce the code churn?

This would mean we have a function `submitScores` that pushes into `setScoresCalls`. That doesn't make sense. It's only coherent if we also rename `submitScores` to `setScores` everywhere. I'm ok with that, wdyt?

> I am surprised we don't need to include the roundIndex when submitting scores. Could the lack of roundIndex make debugging more difficult for us?

`spark-rewards` has no notion of rounds. What would we need the round index for?

> How do you feel about keeping the setScores signature in the interfaces, i.e. modify submitScores to require (roundIndex, participantAddresses, scores), even if the actual implementation calling spark-rewards does not use the round index?

We don't need that right now so I'm against it and think we shouldn't spend time discussing it. If you feel strongly I'll just change it.

Reply:
> > Can you preserve the name setScoresCall to reduce the code churn?
> 
> This would mean we have a function `submitScores` that pushes into `setScoresCalls`. That doesn't make sense. It's only coherent if we also rename `submitScores` to `setScores` everywhere. I'm ok with that, wdyt?

_rename `submitScores` to `setScores` everywhere_ sounds good to me 👍🏻 

> > I am surprised we don't need to include the roundIndex when submitting scores. Could the lack of roundIndex make debugging more difficult for us?
> 
> `spark-rewards` has no notion of rounds. What would we need the round index for?

For troubleshooting - e.g. to detect a duplicate submission of scores for the same round, or a round with no scores submitted.

It also allows us to implement idempotency in spark-rewards: we can add pRetry to spark-evaluate and drop duplicate scores for the same round on the spark-rewards side.

> > How do you feel about keeping the setScores signature in the interfaces, i.e. modify submitScores to require (roundIndex, participantAddresses, scores), even if the actual implementation calling spark-rewards does not use the round index?
> 
> We don't need that right now so I'm against it and think we shouldn't spend time discussing it. If you feel strongly I'll just change it.

Ack!

Reply:
> rename submitScores to setScores everywhere sounds good to me 👍🏻

Done!

> For troubleshooting - e.g. to detect a duplicate submission of scores for the same round, or a round with no scores submitted.
>
> It also allows us to implement idempotency in spark-rewards: we can add pRetry to spark-evaluate and drop duplicate scores for the same round on the spark-rewards side.

I opened a new issue for this, I don't see it as release blocking: https://github.com/filecoin-station/spark-rewards/issues/15